python/mxnet/gluon/data/dataloader.py

```diff
@@ -244,6 +270,35 @@ def __len__(self):
     def __del__(self):
         self.shutdown()
 
+    def reset(self):
```

Comment:
assert not shutdown

Reply:
Do we need to introduce a new reset method? Just moving this functionality into `__iter__` of this class would be cleaner I think